dashboard/cron.py

```diff
@@ -103,13 +110,17 @@ def verify_course_ids(self):
             logger.debug(course_sql)
             course_df = pd.read_sql(course_sql, conns['DATA_WAREHOUSE'])
 
-            # error out when course id is invalid
+            # error out when course id is invalid, otherwise accumulate dataframes
             if course_df.empty:
                 logger.error(f"""Course {course_id} don't have the entry in data warehouse yet. """)
                 invalid_course_id_list.append(course_id)
+            else:
+                course_dfs.append(course_df)
 
+        courses_data = pd.concat(course_dfs).reset_index()
```

Comment:
When no course is added to the app then code should handle it, currently throwing an exception. A check should be made on emptiness of the list `course_dfs`
`raise ValueError(\'No objects to concatenate\')\nValueError: No objects to concatenate\n', None`

Reply:
I will review the code once this error is fixed

Reply:
Thanks for this catch!  I made some modifications in this [commit](https://github.com/tl-its-umich-edu/my-learning-analytics/pull/666/commits/995204b8ad40d8c1277f014ece456057b54dc6bb) that I believe have resolved this. Now you should be able to run `cron.py` with no courses in the database. Essentially, it will update the `academic_terms` table and nothing else.